lib/chef/provider/service/systemd.rb

```diff
@@ -77,7 +77,7 @@ def define_resource_requirements
 
   def get_systemctl_options_args
     if new_resource.user
-      uid = Etc.getpwuid(new_resource.user).uid
+      uid = Etc.getpwnam(new_resource.user).uid
```

Comment:
This'll explode if users have previously been using UIDs here:
```
irb(main):003:0> Etc.getpwnam 501
Traceback (most recent call last):
        3: from /Users/thom/.rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):3
        1: from (irb):3:in `getpwnam'
TypeError (no implicit conversion of Integer into String)
```
probably better to call the right thing if we have a string or an int.

Reply:
regardless we should accept both in the API.

there's fairly horrible problems with glibc caching+ruby and when chef creates the username where uids are required in apis like this in order to work around the problems (or else people need to run nscd, but everyone irrationally hates nscd).

Reply:
Any hope of that being a separate task? This was just meant to be a quick fix for something that's currently outright broken. Adding support and accompanying tests for allowing `user` to be a UID in the `service` and `systemd_unit` resources is not something I really want to take on.

Reply:
Yeah, let's file a bug for that and merge this.